modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java

```diff
@@ -77,53 +74,48 @@ public static Object[][] configProvider() {
 
     @Factory(dataProvider = "configProvider")
     public OAuth2ServiceClientCredentialTestCase(TestUserMode userMode) throws Exception {
-
+        super.init(userMode);
         context = new AutomationContext("IDENTITY", userMode);
         this.username = context.getContextTenant().getTenantAdmin().getUserName();
         this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
     }
 
     @BeforeClass(alwaysRun = true)
     public void testInit() throws Exception {
-
-        String backendURL = context.getContextUrls().getBackEndUrl();
-        loginLogoutClient = new LoginLogoutClient(context);
-        logManger = new AuthenticatorClient(backendURL);
-        identityContextUrls = context.getContextUrls();
         tenantInfo = context.getContextTenant();
         userInfo = tenantInfo.getContextUser();
-
-        String sessionCookie = logManger.login(username, userPassword, context.getInstance().getHosts().get("default"));
-        appMgtclient = new ApplicationManagementServiceClient(sessionCookie, backendURL, null);
-        adminClient = new OauthAdminClient(backendURL, sessionCookie);
-        remoteUSMServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
+        restClient = new OAuth2RestClient(serverURL, tenantInfo);
 
         setSystemproperties();
         client = HttpClientBuilder.create().build();
     }
 
     @AfterClass(alwaysRun = true)
     public void atEnd() throws Exception {
-
```

Comment:
Shall we add the missing new line here?

Reply:
Fixed with https://github.com/wso2/product-is/pull/20096